### PR TITLE
If skipping localization don't skip DOS/DRS resolution, plus... WA-351 WA-380

### PR DESCRIPTION
**Pull request by kshakir**
- Removed `martha_v2` response parsing
- When traversing files using a mapper function then mapper does the exclusion
- Use serialized class instead of config string replace for Martha request generation
- Request partial responses from Martha
- Use JDK standard responses for missing file attributes (size=0, time=epoch, hash=None)
- Copy `timeCreated` from DOS/DRS to file attributes
- Martha `read_string()` uses `gsUri` (gs://bucket/name) instead of `bucket` and `name`
- Martha localization uses safer file paths still based on the DOS/DRS URI
- Reading DOS/DRS content uses the config google auth type, not always Bond-or-USA
- Google config auth types support ADC=SA, passing in scopes to ADC
- Google auth type `UserMode` no longer requires config values that it was ignoring
- Allow skipping docker build-and-push by specifying the `CROMWELL_BUILD_PAPI_DOCKER_IMAGE_DRS`
- `papi-v2-usa` backend now ALWAYS uses the USA just like Terra/FC does

### CODE REVIEW

File: centaur/src/main/resources/standardTestCases/drs_usa_jdr.test [hunk 1]
```diff
@@ -1,3 +1,4 @@
+ignore: false
```
Comment (kshakir) [inline centaur/src/main/resources/standardTestCases/drs_usa_jdr.test:1]: TODO: switch to true before merging

File: src/ci/bin/test_papi.inc.sh [hunk 2]
```diff
@@ -82,13 +82,15 @@ cromwell::private::papi::gcr_image_delete() {
 }
 
 cromwell::private::papi::setup_papi_gcr() {
-    if command -v docker; then
+    if [[ -n "${CROMWELL_BUILD_PAPI_DOCKER_IMAGE_DRS:+set}" ]]; then
```
Comment (kshakir) [inline src/ci/bin/test_papi.inc.sh:85]: Leave a 'why' comment on how this speeds up local testing

File: cloud-nio/cloud-nio-impl-drs/src/main/scala/cloud/nio/impl/drs/DrsPathResolver.scala [hunk 3]
```diff
@@ -51,51 +51,50 @@ abstract class DrsPathResolver(drsConfig: DrsConfig, httpClientBuilder: HttpClie
...
   * @param timeUpdated The last update time of the object at gsUri
-  * @param bucket The bucket name port of gsUri, for example "bucket"
-  * @param name The object name part of gsUri, for example "12/345"
   * @param gsUri Where the object bytes are stored, possibly using a generated path name such as "gs://bucket/12/345"
   * @param googleServiceAccount The service account to access the gsUri contents
   * @param fileName A possible different file name for the object at gsUri, ex: "gsutil cp gs://bucket/12/345 my.vcf"
   * @param hashes Hashes for the contents stored at gsUri
   */
 final case class MarthaResponse(size: Option[Long],
+                                timeCreated: Option[String],
```
Comment (salonishah11) [inline cloud-nio/cloud-nio-impl-drs/src/main/scala/cloud/nio/impl/drs/DrsPathResolver.scala:96]: Why extract `timeCreated`? AFAIK it isn't used anywhere no?

**Review by salonishah11 — APPROVED** [review]

File: cloud-nio/cloud-nio-impl-drs/src/main/scala/cloud/nio/impl/drs/DrsPathResolver.scala (hunk 3 above)
Comment (kshakir) [reply cloud-nio/cloud-nio-impl-drs/src/main/scala/cloud/nio/impl/drs/DrsPathResolver.scala:96]: `timeUpdated` isn't actually used by Cromwell either, but is on the this "cloud-nio / [alfre](https://github.com/broadinstitute/alfre)-super-early-fork" [SPI](https://en.wikipedia.org/wiki/Service_provider_interface).

Out of my sense-of-consistency, I could either remove `timeUpdated` from the implementation of the SPI, or add `timeCreated` to the implementation. But it didn't make sense to only have one field but not both.

File: CHANGELOG.md [hunk 4]
```diff
@@ -1,5 +1,16 @@
...
+## 54 Release Notes
+
+### `martha_v2` Support Removed
+
+Cromwell now only supports resolving DOS or DRS URIs through [Martha](https://github.com/broadinstitute/martha)'s most
+recent metadata endpoint `martha_v3`, dropping support for Martha's previous metadata endpoint `martha_v2`. To switch to
+the new version of Martha's metadata endpoint, update the `martha.url` found in the [filesystems
+config](https://cromwell.readthedocs.io/en/stable/filesystems/Filesystems/#overview) to point to `/martha_v3`. More
+information on Martha's `martha_v3` request and response schema can be found
+[here](https://github.com/broadinstitute/martha#martha-v3).
```
Comment (aednichols) [inline CHANGELOG.md:12]: Can we also say that we added support for DRS with `localization_optional`?

**Review by rsasch — APPROVED** [review]

**Review by aednichols — APPROVED** [review]
I appreciate that you always leave the campground better than you found it.

**Pull request merged**